osu.Game.Tests/Visual/UserInterface/TestSceneNowPlayingOverlay.cs

```diff
@@ -86,11 +102,32 @@ public void TestBackgroundScrolling(bool goNext)
             AddAssert("More than one in playlist", () => musicController.BeatmapSets.Count() > 1);
             AddStep(@"show", () => nowPlayingOverlay.Show());
             AddUntilStep("Is Visible", () => nowPlayingOverlay.Alpha > 0);
-            if (goNext)
-                AddStep("next track", () => musicController.NextTrack());
-            else
-                AddUntilStep("previous track", () => musicController.PreviousTrack() != PreviousTrackResult.Restart);
-            AddAssert("Background is scrolling", () => nowPlayingOverlay.ChildrenOfType<BufferedContainer>().First().Transforms.Any());
+
+            BufferedContainer currentBackground;
+            void step()
+            {
+                currentBackground = getFirstBackground();
+                if (goNext)
+                    AddStep("next track", () => musicController.NextTrack());
+                else
+                    AddUntilStep("previous track", () => musicController.PreviousTrack() != PreviousTrackResult.Restart);
+            }
+
+            step();
+            AddAssert("Background changed", () => getFirstBackground() != currentBackground);
+            step();
+            AddAssert("Previous background is present", () => currentBackground.IsPresent);
```

Comment:
idk why this fails even though you can clearly see the previous background still scrolling by

Reply:
I was assigning value I wanted to test outside a step. Got the tests working in visual now.